app/components/post_draft/draft_input/index.tsx

```diff
@@ -133,16 +133,102 @@ export default function DraftInput({
     const intl = useIntl();
     const serverUrl = useServerUrl();
     const theme = useTheme();
+    const [isEmojiPickerOpen, setIsEmojiPickerOpen] = useState(false);
+    const [isEmojiPickerFocused, setIsEmojiPickerFocused] = useState(false);
 
     const handleLayout = useCallback((e: LayoutChangeEvent) => {
         updatePostInputTop(e.nativeEvent.layout.height);
     }, []);
 
+    // To keep track of cursor position and to retrieve the latest value of cursor position
+    const cursorPositionRef = useRef(cursorPosition);
+    useEffect(() => {
+        cursorPositionRef.current = cursorPosition;
+    }, [cursorPosition]);
+
+    const handleEmojiPress = (emojiName: string) => {
+        updateValue((v) => {
+            const name = emojiName.trim();
+            const currentCursorPosition = cursorPositionRef.current;
+            let unicode;
+            const imageUrl = '';
+            if (EmojiIndicesByAlias.get(name)) {
+                const emoji = Emojis[EmojiIndicesByAlias.get(name)!];
+                if (emoji.category === 'custom') {
+                    return `${v.slice(0, currentCursorPosition)} :${emojiName}: ${v.slice(currentCursorPosition)}`;
+                }
+                unicode = emoji.image;
+                if (unicode && !imageUrl) {
```

Comment:
You are setting `imageUrl` as a constant to  the empty string. This is always going to be true, so you can remove it. If this is copy pasted from a different place, I wonder if it could be refactored into a hook or util function.

Reply:
Yup, we can remove `imageUrl`, I have copy pasted it but with some refactored. I forget it to remove if from the condition. Nice catch 🙌 .